From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Wed, 26 Aug 2026 14:05:00 +0200
Subject: [PATCH] remoteproc: qcom_q6v5_wcss: fix IPQ6018 SSR name

The IPQ6018 SSR-name workaround was refreshed for Linux 6.18 with its
second hunk pointing at the IPQ8074 data instead of IPQ6018.  Consequently
IPQ8074 was changed to "wcnss", while IPQ6018 retained the invalid
"q6wcss" name.

Sending that name to RPM while stopping WCSS resets the SoC, after which a
warm boot hangs in SBL1 Pre_DDR_clock_init.  Restore the IPQ8074 value and
apply the intended "wcnss" value to IPQ6018.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 drivers/remoteproc/qcom_q6v5_wcss.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/remoteproc/qcom_q6v5_wcss.c
+++ b/drivers/remoteproc/qcom_q6v5_wcss.c
@@ -1170,7 +1170,7 @@ static const struct wcss_data wcss_ipq80
 	.crash_reason_smem = WCSS_CRASH_REASON,
 	.aon_reset_required = true,
 	.wcss_q6_reset_required = true,
-	.ssr_name = "wcnss",
+	.ssr_name = "q6wcss",
 	.ops = &q6v5_wcss_ipq8074_ops,
 	.requires_force_stop = true,
 	.need_mem_protection = true,
@@ -1184,7 +1184,7 @@ static const struct wcss_data wcss_ipq60
 	.crash_reason_smem = WCSS_CRASH_REASON,
 	.aon_reset_required = true,
 	.wcss_q6_reset_required = true,
-	.ssr_name = "q6wcss",
+	.ssr_name = "wcnss",
 	.ops = &q6v5_wcss_ipq8074_ops,
 	.requires_force_stop = true,
 	.need_mem_protection = true,
